src/ofxstatement/plugins/germany_1822direkt.py

```diff
@@ -8,8 +8,9 @@
 class FrankfurterSparkasse1822Plugin(Plugin):
     def get_parser(self, filename):
         encoding = self.settings.get('charset', 'iso-8859-1')
-        f = open(filename, 'r', encoding=encoding)
-        parser = FrankfurterSparkasse1822Parser(f)
+        with open(filename, 'r', encoding=encoding) as f:
+            lines = f.readlines()
+        parser = FrankfurterSparkasse1822Parser(lines)
         parser.statement.account_id = self.settings['account']
         parser.statement.bank_id = self.settings.get('bank', '50050201')
         parser.statement.currency = self.settings.get('currency', 'EUR')
```

Comment:
You are right, this is as bug. But honestly, I do not like the fix, because it forces all data into the memory.

I wonder if this would be better ....
```.py
class AutoClose:
    def __init__(self, f):
        self.__f = f
    def __del__(self):
        self.__f.close()
    def __getattr__(self, name):
        return getattr(self.__f, name)

...

    f = AutoClose(open(....))

```

Do you know how other parsers are handling this, I did not check.

Reply:
I agree that this is a downside but nowadays I usually assume that memory is not a problem (with GBs and GBs of it) unless proven otherwise. Premature optimization and so on ;-) Who has bank statements that are several hundred MB big? This program will be executed once and then exit. It won't run as a daemon in the background...

I think `__del__` is not guaranteed to be called or may even be called multiple times.

I think my solution is quite straightforward and explicit, which is also an advantage.

Hopefully the issue I opened in the ofxstatement project will get adressed.

I didn't check how other plugins are solving this issue.

Reply:
Well, You are right. `__del__` is not guaranteed to be called. I think I need to write test for this plugin.